test/suites/standard/cleanup-data.robot

```diff
@@ -0,0 +1,158 @@
+
+Resource            ../../resources/systemd.resource
+Resource            ../../resources/ostree.resource
+Resource            ../../resources/ostree-health.resource
+Resource            ../../resources/microshift-config.resource
+
+Suite Setup         Setup Suite
+
+Test Tags           slow
+
+
+*** Test Cases ***
+Verify Invalid Command Line
+    [Documentation]    Verify invalid command line combinations
+
+    # Usage message
+    ${rc}=    Run MicroShift Cleanup Data    ${EMPTY}
+    Should Not Be Equal As Integers    ${rc}    0
+
+    # Invalid option combination
+    ${rc}=    Run MicroShift Cleanup Data    --ovn    --keep-images
+    Should Not Be Equal As Integers    ${rc}    0
+
+    ${rc}=    Run MicroShift Cleanup Data    --all    --ovn
+    Should Not Be Equal As Integers    ${rc}    0
+
+    ${rc}=    Run MicroShift Cleanup Data    --keep-images
+    Should Not Be Equal As Integers    ${rc}    0
+
+Verify Full Cleanup Data
+    [Documentation]    Verify full data clean scenarios
+
+    ${rc}=    Run MicroShift Cleanup Data    --all
+    Should Be Equal As Integers    ${rc}    0
+    Verify Crio Resource Count    ${TRUE}
+    Verify MicroShift Data Directory And Processes    ${TRUE}
+
+    [Teardown]
+    ...    Start MicroShift And Wait Until Ready
+
+Verify Partial Cleanup Data
+    [Documentation]    Verify partial data clean scenarios
+
+    ${rc}=    Run MicroShift Cleanup Data    --all    --keep-images
+    Should Be Equal As Integers    ${rc}    0
+    Verify Crio Resource Count    ${FALSE}
+    Verify MicroShift Data Directory And Processes    ${TRUE}
+
+    ${rc}=    Run MicroShift Cleanup Data    --keep-images    --all
+    Should Be Equal As Integers    ${rc}    0
+    Verify Crio Resource Count    ${FALSE}
+    Verify MicroShift Data Directory And Processes    ${TRUE}
+
+    Start MicroShift And Wait Until Ready
+
+    ${rc}=    Run MicroShift Cleanup Data    --ovn
+    Should Be Equal As Integers    ${rc}    0
+    Verify Crio Resource Count    ${FALSE}
+    Verify MicroShift Data Directory And Processes    ${FALSE}
+
+    [Teardown]
+    ...    Start MicroShift And Wait Until Ready
+
+
+*** Keywords ***
+Setup Suite
+    [Documentation]    Set up all of the tests in this suite
+    Check Required Env Variables
+    Login MicroShift Host
+    Start MicroShift And Wait Until Ready
+
+Start MicroShift And Wait Until Ready
+    [Documentation]    Start the service and wait until full initialized
+    Systemctl    enable    --now microshift
+    Restart Greenboot And Wait For Success
+
+Run MicroShift Cleanup Data
+    [Documentation]    Run the microshift-cleanup-data script and
+    ...    return its exit code
+    [Arguments]    ${cmd}    ${opt}=${EMPTY}
+    ${stdout}    ${stderr}    ${rc}=    Execute Command
+    ...    echo 1 | sudo microshift-cleanup-data ${cmd} ${opt}
+    ...    return_stdout=True    return_stderr=True    return_rc=True
+    RETURN    ${rc}
+
+Verify Crio Resource Count
```

Comment:
If this was written as an assertion, I think it would be obvious it needs to be split into a few keywords.

- Crio should have 1 pod
- Crio should have 1 process
- Crio should have 1 image

Then the caller would use those instead of this jumbo keyword, and instead of an argument that triggers the conditional check on line 109 there would be a separate keyword for

- Crio should have 0 images